### PR TITLE
Mypy typing introduction

### PR DESCRIPTION
**Pull request by 3asm**

### CODE REVIEW

File: src/ostorlab/agent/agent.py [hunk 1]
```diff
@@ -275,7 +270,7 @@ def main(cls, args=None) -> NoReturn:
         connect to. Some of these settings are consumed by the scan runtime, others are consumed by the agent itself.
 
         Args:
-            args: Arguments passed to the argument parser. These are added for testability.
+            parsed_args: Arguments passed to the argument parser. These are added for testability.
```
Comment (mohsinenar) [inline src/ostorlab/agent/agent.py:273]: Where is `parsed_args` coming from ?

Comment (3asm) [inline src/ostorlab/agent/agent.py:273]:
```suggestion
            args: Arguments passed to the argument parser. These are added for testability.
```

File: .github/workflows/python-package-lint.yml [hunk 2]
```diff
@@ -30,9 +30,12 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         python -m pip install pylint pylint-quotes
+        python -m pip install mypy types-click types-protobuf types-redis types-requests types-urllib3 typing_extensions
```
Comment (deadly-panda) [inline .github/workflows/python-package-lint.yml:33]: why u dont want to install the types with mypy automatically ?

File: src/ostorlab/assets/android_aab.py [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
 
 
-@dataclasses.dataclass
 @asset.selector('v3.asset.file.android.aab')
 class AndroidAab(asset.Asset):
     """Android .AAB target asset."""
-    content: bytes
-    path: Optional[str] = None
+
+    def __init__(self, content: bytes, path: Optional[str]):
```
Comment (deadly-panda) [inline src/ostorlab/assets/android_aab.py:11]: default path to None if it is Optional.

File: src/ostorlab/assets/ipv4.py [hunk 4]
```diff
@@ -9,9 +9,11 @@
 @asset.selector('v3.asset.ip.v4')
 class IPv4(asset.Asset):
     """IPv4 Address target asset."""
-    host: str
-    version: int = 4
-    mask: Optional[str] = None
+
+    def __init__(self, host: str, version: Optional[int] = 4, mask: Optional[str] = None):
```
Comment (deadly-panda) [inline src/ostorlab/assets/ipv4.py:13]: Shouldn't the default depend on the host ?

File: tests/agent/samples/process_agent.py [hunk 5]
```diff
@@ -2,8 +2,8 @@
 import datetime
 import logging
 
-from ostorlab.agent import agent, message as agent_message
-
+from ostorlab.agent import message as agent_message
```
Comment (deadly-panda) [inline tests/agent/samples/process_agent.py:5]:
```suggestion
from ostorlab.agent.message import message as agent_message
```

File: tests/agent/samples/start_agent.py [hunk 6]
```diff
@@ -1,9 +1,8 @@
 """Sample agent that implements the start method and sends a message."""
 import datetime
 
-from ostorlab.agent import agent
+from ostorlab.agent import agent, definitions as agent_definitions
```
Comment (deadly-panda) [inline tests/agent/samples/start_agent.py:4]:
```suggestion
from ostorlab.agent import agent
from ostorlab.agent import definitions as agent_definitions
```

**Review by deadly-panda — APPROVED** [review]

File: src/ostorlab/assets/android_aab.py (hunk 3 above)
Comment (deadly-panda) [inline src/ostorlab/assets/android_aab.py:11]:
```suggestion
    def __init__(self, content: bytes, path: Optional[str] = None):
```

File: .github/workflows/python-package-lint.yml (hunk 2 above)
Comment (3asm) [reply .github/workflows/python-package-lint.yml:33]: --install-types with --no-interactive didn't install deps we had. Also not a big fan of letting mypy decide what to install.

**Pull request merged**